### PR TITLE
Tramming wizard usability improvements

### PR DESCRIPTION
**Pull request by sjasonsmith**
### Description

This is a continuation of the following PR. Despite it being labeled as Merged, it is not yet in bugfix-2.0.x.

This makes two changes to ASSISTED_TRAMMING_WIZARD:
1. Allow any point to be probed first and become the reference point. Previously you had to start with the first point in the list.
2. Do not show "Last value" if the point has not yet been probed. Previously you would see non-zero values here before probing the point, because "reference - 0.0" resulted in a non-zero number.

On a mega2560, these changes increase RAM usage by 5 bytes and FLASH usage by 92 bytes.

### Requirements

ASSISTED_TRAMMING_WIZARD

### Benefits

Somewhat more intuitive usage.

### Configurations

These are for a very non-stock MakerGear M2 with these features enabled:
[Modified_M2.zip](https://github.com/MarlinFirmware/Marlin/files/7071381/Modified_M2.zip)

### Related Issues

#22201

### CODE REVIEW

**Comment by sjasonsmith** [comment]
@thinkyhead this is now tested.

I had to make two changes on top of your commit to get it to build and work properly:

1. The compiler didn't like the local variable being used in the lambda. Perhaps this could have been resolved with some capture options, but I know if the past I've had trouble with capture flexibility since we don't have the full STL available to us.

2. The logic to decide whether to display the number or not lost the check for a valid reference value. **Now I'm questioning why I had to make this change, since it shouldn't be possible to have a valid tram_index without a valid reference_index**. I'll re-test that and figure out what was wrong.

**Comment by sjasonsmith** [comment]
OK, the second commit wasn't needed and has been reverted. I had made that change before fixing another issue in the previous commit.

File: Marlin/src/lcd/menu/menu_tramming.cpp [hunk 1]
```diff
@@ -50,32 +52,32 @@ static bool probe_single_point() {
...
 
 static void tramming_wizard_menu() {
-  DEBUG_ECHOLNPAIR("Screen: tramming_wizard_menu");
   START_MENU();
   STATIC_ITEM(MSG_SELECT_ORIGIN);
 
   // Draw a menu item for each tramming point
   LOOP_L_N(i, G35_PROBE_COUNT)
-    SUBMENU_N_P(i, (char*)pgm_read_ptr(&tramming_point_name[i]), []{ _menu_single_probe(MenuItemBase::itemIndex); });
+    SUBMENU_N_P(i, (char*)pgm_read_ptr(&tramming_point_name[i]), _menu_single_probe);
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu_tramming.cpp]: I'm surprised that the value of `i` is not available to be passed in `MenuItemBase::itemIndex` as this construct is used in many places. For example…
```cpp
LOOP_L_N(n, EXTRUDERS)
  EDIT_ITEM_N(int3, n, MSG_FLOW_N, &planner.flow_percentage[n], 10, 999, []{ planner.refresh_e_factor(MenuItemBase::itemIndex); });
```
I see that you decided to follow the same pattern as `CANCEL_OBJECT`, picking up the index in the called handler above. That should work. However, I would definitely prefer to standardize and figure out a more elegant way to pass the index directly.

Comment (sjasonsmith) [reply Marlin/src/lcd/menu/menu_tramming.cpp]: I think the key difference is that in the working example it is passing a static variable and not a local.

I didn’t spend too much time fighting it. If you can get it to build that’s great, but it wouldn’t compile how it was.

Removing the argument from that function call actually decreased FLASH usage by a few dozen bytes as well.

Comment (sjasonsmith) [reply Marlin/src/lcd/menu/menu_tramming.cpp]: It was actually failing on the single point menu, where it was trying to add a menu item that was a local function argument.

Comment (sjasonsmith) [reply Marlin/src/lcd/menu/menu_tramming.cpp]: You might be able to just explicitly capture that by value. I know I’ve hit limitations in the past on using lambdas without the full STL available, but I can’t remember what the limitations were.

File: Marlin/src/lcd/menu/menu_tramming.cpp [hunk 2]
```diff
@@ -50,32 +52,32 @@ static bool probe_single_point() {
...
-  return !isnan(z_probed_height);
+  DEBUG_ECHOLNPAIR("probe_single_point(", tram_index, ") = ", z_probed_height, "mm");
+  return (z_isvalid[tram_index] = !isnan(z_probed_height));
 }
 
-static void _menu_single_probe(const uint8_t point) {
-  tram_index = point;
-  DEBUG_ECHOLNPAIR("Screen: single probe screen Arg:", point);
+static void _menu_single_probe() {
+  tram_index = MenuItemBase::itemIndex;
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu_tramming.cpp]: The tricky part here is that if `_menu_single_probe` were to contain any of the menu item macros that write to this index, it and `tram_index` will get trounced, and a future reviewer may not realize this is why `tram_index` is getting stepped on. Something to watch out for.

File: Marlin/src/lcd/menu/menu_tramming.cpp (hunk 1 above)
Comment (thinkyhead) [reply Marlin/src/lcd/menu/menu_tramming.cpp]: Right, because `MenuItemBase::itemIndex` is static and global.

**Pull request merged**